### PR TITLE
Merge Function

### PR DESCRIPTION
**Pull request by robbieyang**
basi merge function added.

### CODE REVIEW

**Review by wai2016 — COMMENTED** [review]
The spacing is not good so the code is hard to read XD

File: app/team.html [hunk 1]
```diff
@@ -46,87 +46,104 @@ <h2 id="text_event_name"></h2>
...
-					 <div ng-repeat="r in requests">
-						<button type="button" class="btn" ng-click="processRequest(r);">
-							<span class="glyphicon glyphicon-backward" aria-hidden="true"></span> {{r}}
-						</button>
-					 </div>
-					 
-                  </div>
+			<div class="col-sm-3">
+				<h4>Requests received:</h4>
+				<div ng-repeat="r in requests">
```
Comment (wai2016) [inline app/team.html]: I think add back the comment `<!--Show the requests -->` is good.

**Comment by robbieyang** [comment]
Oh, in fact the original code is not well structured and I put not enough comments on it.
I plan to rebuild the code in following days. Sorry for that.
Btw, I would like to suggest that if you are taking charge of one of the features, probably you may spend some time on re-structuring the existing codes to make sure the code readable. Thanks.

File: app/js/team.js [hunk 2]
```diff
@@ -156,11 +181,42 @@ angular.module('teamform-team-app', ['firebase'])
...
 
 	}
+
+		/*************** TeamMergeRequest *******************/
+	$scope.mergeSelection = [];
+	$scope.teamMergeRequest = function(teamItem){
+		if(typeof teamItem.teamMembers == "undefined"){
+			teamItem.teamMembers = [];
+		}
+		if($scope.param.teamMembers.length + teamItem.teamMembers.length <= teamItem.size && $scope.mergeSelection.length < 1){
```
Comment (wai2016) [inline app/js/team.js:200]: Is that `$scope.mergeSelection.length < 1` controlling team leader to send request to only one team?

File: app/team.html [hunk 3]
```diff
@@ -46,87 +46,104 @@ <h2 id="text_event_name"></h2>
...
+					</button>
 				</div>
-				  
-				  
+			</div>
+
+			<div class="col-sm-3">
+				<h4>Your preferred team to merge</h4>
+				<div ng-repeat="t in param.teamToSelect " ng-show="t.$id != param.teamName">
+					<button type="button" class="btn" ng-click="teamMergeRequest(t);">
```
Comment (wai2016) [inline app/team.html:134]: I think there should be a button here for cancelling the merge request after sending the request.

**Review by wai2016 — COMMENTED** [review]
There should more comments. XD
Because it's hard for me to understand your code in a short time.
And I have added some recommendation there.

File: app/team.html [hunk 4]
```diff
@@ -46,87 +46,104 @@ <h2 id="text_event_name"></h2>
...
+						<span class="glyphicon glyphicon-remove" aria-hidden="true"></span> {{member}}
+					</button>
+				</div>
+                     
+			</div>
 
-                  <div class="col-sm-6">
-                     <h4>Requests received:</h4>
-
-                     <!--TODO: Show the requests -->
```
Comment (wai2016) [inline app/team.html:115]: I think add back the comment `<!--Show the requests -->` is good.

File: app/js/team.js (hunk 2 above)
Comment (robbieyang) [reply app/js/team.js:200]: Yes, each team should have at most 1 merge request at the same time.

Comment (p-wong1124) [reply app/js/team.js:200]: For both receive side and send side??

**Comment by wai2016** [comment]
I think it's ok to merge.

**Comment by katrinako** [comment]
Sooo great!! You guys are genius~

**Comment by RaymondWongCH** [comment]
I think it can function well. It is ok to merge.

**Pull request merged**